internal/pkg/plugin/manager.go

```diff
@@ -577,38 +580,30 @@ func (m *manager) SendersStatus() map[string]SenderStatus {
 }
 
 func (m *manager) UnregisterSender(ctx context.Context, ps pkgsender.Sender) error {
-
 	s, ok := ps.(*sender)
 	if !ok || !s.active {
 		return &RegistrationError{
 			Message: "sender not registered",
 		}
 	}
-
-	key := m.mapkey(s.name, s.hash)
-
+	key := m.mapkey(s.tid, s.name, s.hash)
 	m.Lock()
-
 	m.sendersCount[key]--
-
 	if m.sendersCount[key] <= 0 {
 		delete(m.sendersCount, key)
 		delete(m.senders, key)
 	}
-
 	delete(m.sendersWrapped, s.id)
 	m.Unlock()
-
 	s.Lock()
 	s.active = false
 	s.Unlock()
-
 	return nil
 
 }
 
 // === Helper Functions ==============================================
 
-func (m *manager) mapkey(name string, hash string) string {
-	return name + "/" + hash
+func (m *manager) mapkey(tid tenant.Id, name string, hash string) string {
+	return tid.OrgId + "/" + tid.AppId + "/" + name + "/" + hash
```

Comment:
Why are we not using tid.Key() function? If we agree that we want to allow all characters in orgId and appId, we should stick to a more generic key generation.

Reply:
Got it. Didn't realize that this just need to be one way. I take back my comment. This is ok

Reply:
I just realize that {"orgId": "foo/foo", "appId":"bar"} will result in the same mapkey as {"orgId":"foo", appId:"foo/bar"}. Granted, this is an edge case, but we can no longer claim that all characters are allowed